receiver.go

```diff
@@ -133,49 +134,72 @@ func (d *receiveHandler) Receive(length int) error {
 }
 
 func (r *receiveHandler) receiveNotify() (err error) {
-	req := newRequest(MethodNotify)
+	req := r.newRequest(MethodNotify)
 	return r.handleReceiveDispatch(req)
 }
 
 func (r *receiveHandler) receiveCall() error {
-	req := newRequest(MethodCall)
+	req := r.newRequest(MethodCall)
 	return r.handleReceiveDispatch(req)
 }
 
 func (r *receiveHandler) receiveCancel() (err error) {
-	req := newRequest(MethodCancel)
-	if err := decodeIntoMessage(r.reader, req.Message()); err != nil {
+	req := r.newRequest(MethodCancel)
+	m := req.Message()
+	if err := decodeIntoMessage(r.reader, m); err != nil {
+		return err
+	}
+	if err := decodeToNull(r.reader, m); err != nil {
 		return err
 	}
-	req.LogInvocation(r.log, nil, nil)
+	req.LogInvocation(nil, nil)
 	r.taskCancelCh <- req.Message().seqno
 	return nil
 }
 
 func (r *receiveHandler) handleReceiveDispatch(req request) error {
-	if err := decodeIntoMessage(r.reader, req.Message()); err != nil {
+	m := req.Message()
+	if err := decodeIntoMessage(r.reader, m); err != nil {
 		return err
 	}
 
-	m := req.Message()
+	// Handle the dispatched call
 	serveHandler, wrapErrorFunc, se := r.findServeHandler(m.method)
 	if se != nil {
+		fmt.Printf("Server error: %v\n", se)
```

Comment:
is this debug code you forgot to remove? If not, don't we have a logger or something?